src/Compilers/Core/Portable/InternalUtilities/OneOrMany.cs

```diff
@@ -184,6 +186,41 @@ public IEnumerable<TResult> OfType<TResult>()
             return default;
         }
 
+        public static OneOrMany<T> CastUp<TDerived>(OneOrMany<TDerived> from) where TDerived : class, T
+        {
+            return from.HasOne
+                ? new OneOrMany<T>(from._one)
+                : new OneOrMany<T>(ImmutableArray<T>.CastUp(from._many));
+        }
+
+        public bool All(Func<T, bool> predicate)
+        {
+            foreach (var value in this)
+            {
+                if (!predicate(value))
+                    return false;
+            }
+
+            return true;
+        }
+
+        public bool Any()
+            => this.Count > 0;
+
+        public bool Any<TData>(Func<T, TData, bool> predicate, TData data)
+        {
+            foreach (var value in this)
+            {
+                if (predicate(value, data))
+                    return true;
+            }
+
+            return false;
+        }
```

Comment:
Are these methods used?

Reply:
Yes. By existing callers of similar methods that used to operate on an ImmutableArray.